packages/quantic/cypress/integration/facets/timeframe-facet/timeframe-facet.cypress.ts

```diff
@@ -0,0 +1,365 @@
+        }
+        res.send();
+      });
+    }).as(InterceptAliases.Search.substring(1));
+  }
+
+  describe('with empty timeframes', () => {
+    it('should display non-empty timeframes only', () => {
+      mockSearchEmptyTimeframes();
+      cy.visit(pageUrl);
+      configure({});
+      cy.wait(InterceptAliases.Search);
+
+      scope('with one non-empty timeframe', () => {
+        Expect.numberOfValues(1);
+      });
+    });
+  });
+
+  describe('with non-empty timeframes', () => {
+    it('should display timeframes in the right order', () => {
+      mockSearchWithAllTimeframes();
+      cy.visit(pageUrl);
+      configure({});
+      cy.wait(InterceptAliases.Search);
+
+      Expect.valuesEqual([
+        'Next year',
+        'Past 2 weeks',
+        'Past month',
+        'Past year',
+        'Last decade',
+      ]);
+    });
+  });
+
+  describe('with with-date-picker', () => {
+    it('should activate manual range', () => {
+      visitTimeframeFacet({
+        withDatePicker: true,
+      });
+
+      scope('validate initial rendering', () => {
+        Expect.displayLabel(true);
+        Expect.displayCollapseButton(true);
+        Expect.displayStartInput(true);
+        Expect.displayEndInput(true);
+        Expect.displayApplyButton(true);
+        Expect.displayValues(true);
+      });
+
+      scope('when collapsing the facet', () => {
+        Actions.collapse();
+
+        Expect.displayStartInput(false);
+        Expect.displayEndInput(false);
+        Expect.displayApplyButton(false);
+
+        Actions.expand();
+      });
+
+      scope('when specifying a range', () => {
+        Actions.applyRange('2000-12-31', '2001-03-15');
+
+        cy.wait(InterceptAliases.Search);
+
+        const range = '2000/12/31@00:00:00..2001/03/15@23:59:59';
+        Expect.urlHashContains('Date_input', range);
+        Expect.logSelectedValue('Date', range);
+
+        Expect.displayValues(false);
+        Expect.displayClearButton(true);
+      });
+
+      scope('when clearing filter', () => {
+        Actions.clearFilter();
+
+        cy.wait(InterceptAliases.Search);
+
+        Expect.displayClearButton(false);
+        Expect.displayValues(true);
+        Expect.urlHashIsEmpty();
+        Expect.logClearFilter('Date');
+      });
+
+      scope('when entering an invalid range', () => {
+        scope('invalid start date format', () => {
+          Actions.applyRange('bad start date', '2000-01-01');
+
+          Expect.validationError(
+            'Your entry does not match the allowed format yyyy-MM-dd.'
+          );
+        });
+
+        scope('invalid end date format', () => {
+          Actions.applyRange('2000-01-01', 'bad end date');
+
+          Expect.validationError(
+            'Your entry does not match the allowed format yyyy-MM-dd.'
+          );
+        });
+
+        scope('end date smaller than start date', () => {
+          Actions.applyRange('2000-12-31', '2000-01-01');
```

Comment:
is it worthwhile creating constants for these two dates that seem to be reused often?

Reply:
Yes that's worth it :smile: